plugins/GSdx/stdafx.cpp

```diff
@@ -71,16 +71,56 @@ void vmfree(void* ptr, size_t size)
 	VirtualFree(ptr, 0, MEM_RELEASE);
 }
 
+static HANDLE s_fh = NULL;
+static uint8* s_Next[8];
+
 void* fifo_alloc(size_t size, size_t repeat)
 {
-	// FIXME check linux code
-	return vmalloc(size * repeat, false);
+	ASSERT(s_fh == NULL);
+
+	if (repeat >= countof(s_Next)) {
+		fprintf(stderr, "Memory mapping overflow (%zu >= %u)\n", repeat, countof(s_Next));
+		return nullptr;
+	}
+
+	s_fh = CreateFileMapping(INVALID_HANDLE_VALUE, nullptr, PAGE_READWRITE, 0, size * repeat, nullptr);
+	DWORD errorID = ::GetLastError();
+	if (s_fh == NULL) {
+		fprintf(stderr, "Failed to reserve memory. WIN API ERROR:%u\n", errorID);
+		return nullptr;
+	}
+
+	void* fifo = MapViewOfFile(s_fh, FILE_MAP_ALL_ACCESS, 0, 0, size);
+
+	for (size_t i = 1; i < repeat; i++) {
+		void* base = (uint8*)fifo + size * i;
+		s_Next[i] = (uint8*)MapViewOfFileEx(s_fh, FILE_MAP_ALL_ACCESS, 0, 0, size, base);
```

Comment:
From the documentation:

> If the lpBaseAddress parameter specifies a base offset, the function succeeds if the specified memory region is not already in use by the calling process.

So if this call fails some random memory that is already in use might get overwritten.